### ISSUE TITLE
Avoid formatting the HTML when saving

### ISSUE THREAD

**Issue by carpusmedia** [GetmeUK/ContentTools#266]
Hello,

Thank you for the best inline editor in this world.

I have a problem. When I save, line-breaks are added between the opening HTML tag and the text content. As I style these elements with `white-space: pre-line;`, I'd like the white-space between the opening HTML tag and the text content to remain untouched.

Is there anyway to instruct ContentTools not to change the white-space/line-breaks for these elements when saving?

**Comment by anthonyjb** [comment]
Hi @carpusmedia - thanks for your kind comments :)

So there's not a set solution for this at the moment, the best you can do is set the indentation to be `''` like so:

```JavaScript
ContentEdit.INDENT = '';
```

But you'd still then be left with the line endings. I'm going to convert this to an enhancement because I think it would be useful to be able to set the line-endings type as well, something like:

```JavaScript
ContentEdit.LINE_ENDING = '';
```

Will add to the next release.

In the meantime the easiest solution is to strip the problematic white spaces from your HTML at the point of save, so client-side you could do this like so (I think haven't tested the regex directly):

```JavaScript
var regions = ev.detail().regions;
for (name in regions) {
    regions[name] = regions[name].replace(/>\s+|\s+</g, '');
}
```

**Comment by carpusmedia** [comment]
Thank you @anthonyjb for such an awesome response, apologize for the delay.

Turned out that the first line-break was the most important to get excluded in order to make the text content respect `white-space: pre-line;`. So I followed your advice and simply added a listener for the `saved` event, like so:

```
editor.addEventListener('start', function (ev) {
  ...
  for (name in regions) {
    // Remove the first line break
   regions[name] = regions[name].replace(/\r?\n|\r/,"");
  }
}
```

This worked out wonderfully.

Thank you again,
Chris
